Model version and details archive

dpanshug opened the pull request:
[RHOAIENG-6639](https://issues.redhat.com/browse/RHOAIENG-6639)
[RHOAIENG-6640](https://issues.redhat.com/browse/RHOAIENG-6640)

## Description
Archiving model versions.

See below video for archiving and restoring flow, along with archive model version table and details view.
[Screencast from 2024-05-20 16-07-17.webm](https://github.com/opendatahub-io/odh-dashboard/assets/97534722/cc36b216-b2eb-41ab-ac24-30d82c129f97)


## How Has This Been Tested?
1. Move to model registry section and open a register model.
2. Try archiving and restoring model version.

## Test Impact
Added unit tests and cypress tests

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

@yih-wang

yih-wang commented:
Thanks @dpanshug for all the work! That looks great!

Several comments:
* After a version is restored, the user should be navigated to the version details page rather than the versions list. That will provide convenience for the following actions to that version.
* Not sure whether that should be fixed here, but in the empty state of the versions list, the link button should be put under the primary button, rather than next to each other. That follows the PF guidelines.
* I've noticed numerous tooltips are appearing when interacting with the interface. Are these tooltips part of the default PF template? It seems like they are too much and not necessary in all cases. Is that possible to display only the tooltips specified in the designs?
<img width="550" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/23331885/60791c6e-fbfa-4ad1-a6bc-a21866e4a9e4">
<img width="456" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/23331885/715c5a33-1e95-4fca-9caf-ab02a7d1da0d">

dpanshug commented:
@yih-wang Pls check the latest screencast video. 

> After a version is restored, the user should be navigated to the version details page rather than the versions list. That will provide convenience for the following actions to that version.

I've updated this so restoring from the details page navigates to the version details page. Should the same happen when restoring from the Archive version table?

> Not sure whether that should be fixed here, but in the empty state of the versions list, the link button should be put under the primary button, rather than next to each other. That follows the PF guidelines.

Corrected this.

> I've noticed numerous tooltips are appearing when interacting with the interface. Are these tooltips part of the default PF template? It seems like they are too much and not necessary in all cases. Is that possible to display only the tooltips specified in the designs?

That's because we are using [Truncate](https://www.patternfly.org/components/truncate) component for the name. I am not sure if we can remove the unwanted tooltip. cc @lucferbux

manaswinidas commented on frontend/src/pages/modelRegistry/screens/routeUtils.ts:
```diff
@@ -1,8 +1,21 @@
+  rmId?: string,
+  preferredModelRegistry?: string,
+): string =>
+  `/modelRegistry/${preferredModelRegistry}/registeredModels/${rmId}/versions/archive/${mvId}`;
```
I think this route can be changed to `archived` instead of `archive`, since it's already archived?

dpanshug replied:
`archive` sounds better to me, as it denotes a list of archived items. @lucferbux, any thoughts on this?

lucferbux replied:
That's something that we'll need to run down with the UX team for microcopy, IMO if it were `archivedModels` i would go with the past but given that it's a state `archive` sounds better.

manaswinidas commented on frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelVersionArchive.cy.ts:
```diff
@@ -0,0 +1,320 @@
+      name: 'model version 1',
+      author: 'Author 1',
+      id: '1',
+      customProperties: {
```
Do we need the `customProperties` here ? If yes, can we move this to another shared file since a lot of mocks use this?

lucferbux commented on frontend/src/pages/modelRegistry/ModelRegistryRoutes.tsx:
```diff
@@ -47,6 +49,29 @@ const ModelRegistryRoutes: React.FC = () => (
             />
             <Route path="*" element={<Navigate to="." />} />
           </Route>
+          <Route path="versions/archive">
```
I'm leaving that here, not just for your PR, maybe I'll create a task for this, but we should start thinking about splitting routes to decouple a little bit this section.

lucferbux commented on frontend/src/pages/modelRegistry/screens/components/RestoreModelVersionModal.tsx:
```diff
@@ -0,0 +1,65 @@
+
+    try {
+      await onSubmit();
+      setIsSubmitting(false);
```
If you end up always setting submit to false use `finally`

lucferbux commented on frontend/src/pages/modelRegistry/screens/ModelVersionsArchive/ModelVersionArchiveDetails.tsx:
```diff
@@ -0,0 +1,127 @@
+      <ApplicationsPage
+        {...pageProps}
+        breadcrumb={
+          <Breadcrumb>
```
This seems long enought to convert in it's own component don't you think?

lucferbux reviewed (changes requested):
There are a couples of nits in the code, haven't had the time to properly test the functionality, will do tomorrow, tests seemed great!

yih-wang commented:
@dpanshug Thanks for all the updates! The new video looks good.

> I've updated this so restoring from the details page navigates to the version details page. Should the same happen when restoring from the Archive version table?

Yes, lets keep both flows ending up with the details page.

> That's because we are using Truncate component for the name. I am not sure if we can remove the unwanted tooltip. cc @lucferbux

What makes sense to me is only the truncated names have hovering tooltips. Not sure whether it's an easy way to estimate whether it's a truncated name or not. Can we remove the tooltip also for the field below? Seems it's not due to the truncate component.
![image](https://github.com/opendatahub-io/odh-dashboard/assets/23331885/ff1209c2-d6ba-4ebd-b735-9aaa34afe4be)

lucferbux commented:
> What makes sense to me is only the truncated names have hovering tooltips. Not sure whether it's an easy way to estimate whether it's a truncated name or not. Can we remove the tooltip also for the field below? Seems it's not due to the truncate component. ![image](https://private-user-images.githubusercontent.com/23331885/332309731-ff1209c2-d6ba-4ebd-b735-9aaa34afe4be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYyNzU3MTgsIm5iZiI6MTcxNjI3NTQxOCwicGF0aCI6Ii8yMzMzMTg4NS8zMzIzMDk3MzEtZmYxMjA5YzItZDZiYS00ZWJkLWI3MzUtOWFhYTM0YWZlNGJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIxVDA3MTAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjODI3M2VhYzNkMTFlM2ZiMWRkOTBlOTUyMjM5Yjc3YzhkZjg1ZTg3NDc4ZjMwZDIyM2ViYzVkMmQzNTk3NDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BTdiINtqHaZeiWAVoBvUkDIk8bqZ2mlw5Br5o3cmw7Q)

@yih-wang @dpanshug Oh i assumed that wasn't a tooltip but a native autofill in the browser, let me review the code but I couldn't recall a tooltip in the confirmation modal

yih-wang commented:
@lucferbux Ah yes, the one in the confirmation modal might be a browser-native component. It looks different from the tooltip above.

lucferbux commented:
> That's because we are using [Truncate](https://www.patternfly.org/components/truncate) component for the name. I am not sure if we can remove the unwanted tooltip. cc @lucferbux
I was assuming we are mostly reusing what we have in other views, if we have tooltips and truncate text in say model serving and pipelines we should stick to that.

yih-wang commented:
> I was assuming we are mostly reusing what we have in other views, if we have tooltips and truncate text in say model serving and pipelines we should stick to that.

Agree. I'm fine if all the other places do the same thing. We could just follow the pattern and improve this as a whole in the future.

manaswinidas reviewed (approved):
/lgtm Thanks for making all the changes.

manaswinidas commented:
/lgtm

lucferbux commented:
/retest

It seems a problem with cypress and the cdn

lucferbux commented:
/retest
